### PR TITLE
Add `AdoptDevice` and `ForgetDevice` methods

### PR DESCRIPTION
**Pull request by joshuaspence**
Adds `AdoptDevice` and `ForgetDevice` methods and expose `adopted` and `state` fields. Device state is an integer and the relevant code from the decompiled JAR describes the possible values:

```
package com.ubnt.data;

public class Device extends X implements Sanitizable
{

    public static String getStateName(final int n) {
        return Device.O0o000[n];
    }

    public Device setState(final int i) {
        final int state = this.getState();
        final boolean containsField = this.containsField("state");
        this.put("state", i);
        if (i == 2) {
            this.put("adopt_status", 0);
        }
        this.o00000(state, !containsField);
        return this;
    }
    
    public int getState() {
        return this.getInt("state", 0);
    }
    
    public static boolean consideredConnected(final int n) {
        return n == 1 || n == 5;
    }
    
    public boolean consideredConnected() {
        return consideredConnected(this.getState());
    }
    
    public static boolean isBusy(final int n) {
        return n == 7 || n == 5 || isUpgrading(n) || n == 8;
    }
    
    public static boolean isHeartBeatMissed(final int n) {
        return n == 6;
    }
    
    public static boolean isUpgrading(final int n) {
        return n == 4;
    }
    
    public boolean isBusy() {
        return isBusy(this.getState());
    }
    
    public static boolean isConnected(final int n) {
        return n == 1;
    }
    
    public boolean isConnected() {
        return isConnected(this.getState());
    }
    
    public static boolean consideredDisconnected(final int n) {
        return !consideredConnected(n) && !isBusy(n);
    }
    
    public boolean consideredDisconnected() {
        return consideredDisconnected(this.getState());
    }

    
    static {
        O0o000 = new String[] { "UNKNOWN", "CONNECTED", "PENDING", "FIRMWARE_MISMATCH", "UPGRADING", "PROVISIONING", "HEARTBEAT_MISSED", "ADOPTING", "DELETING", "INFORM_ERROR", "ADOPT_FAILED", "ISOLATED" };
    }
}
```

### CODE REVIEW

File: unifi/device.go [hunk 1]
```diff
@@ -2,6 +2,22 @@ package unifi
 
 import (
 	"context"
+	"fmt"
+)
+
+const (
```
Comment (joshuaspence) [inline unifi/device.go:8]: I don't need all of these so if you'd prefer I can get rid of the unused constants.

Comment (paultyng) [reply unifi/device.go:8]: If they would be useful for other folks using the SDK I think its fine. I don't know if we have this precedent anywhere, but I think the const's [should still be `MixedCaps`](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps), ie. `DeviceStateUnknown`? But I don't really feel that strongly, so if you do, ok with me.

Comment (paultyng) [reply unifi/device.go:8]: You could also create a type for this to make the usage more obvious, ie:
```go
type DeviceState int

const (
  DeviceStateUnknown DeviceState = 0
  DeviceStateConnected
  DeviceStatePending
  // ...
)

type Device struct {
  // ...
  State DeviceState
  // ...
}
```

**Review by paultyng — APPROVED** [review]
LGTM, added some comments but your choice on implementation.

**Pull request merged**